### DISCUSSION TITLE
How to install extention requrements correctly?

### DISCUSSION THREAD

**Discussion by light-and-ray** [oobabooga/text-generation-webui#4287] [Q&A]
I tried to install 
```
. ./installer_files/conda/bin/activate
pip install -r extensions/superbooga/requirements.txt
```

But it doesn't affect to the ui. I tried all other "activate" files:
```
./installer_files/env/lib/python3.10/venv/scripts/common/activate
./installer_files/conda/lib/python3.10/site-packages/conda/shell/bin/activate
./installer_files/conda/lib/python3.10/venv/scripts/common/activate
./installer_files/conda/pkgs/conda-23.3.1-py310h06a4308_0/lib/python3.10/site-packages/conda/shell/bin/activate
./installer_files/conda/pkgs/conda-23.3.1-py310h06a4308_0/bin/activate
./installer_files/conda/pkgs/python-3.10.13-h955ad1f_0/lib/python3.10/venv/scripts/common/activate
./installer_files/conda/pkgs/python-3.10.10-h7a1cb2a_2/lib/python3.10/venv/scripts/common/activate
./installer_files/conda/bin/activate
```

But only `./installer_files/conda/bin/activate` affects on pip. (Other says `"error: externally-managed-environment"`)

I found the solution, I copied `extensions/superbooga/requirements.txt` into the end of `./requirements.txt`, and then launched `./update_linux.sh`. It helped, but it looks bad

OS: Manjaro

P.s. It is better to install requirements automatically for extensions like it A1111 does

**Comment by missionfloyd** [comment]
None of that should be necessary. Assuming you're running it with `./launch_linux.sh`, dependencies (including those for extensions) are installed automatically.

To manage them manually, you can use `./cmd_linux.sh` to activate the env.

**Reply by light-and-ray** [reply]
> Assuming you're running it with `./launch_linux.sh`

Yes

> dependencies (including those for extensions) are installed automatically.

Unfortunately it didn't happen for me. I was catching no module chromadb found error. Maybe I should reproduce it and open an issue

> To manage them manually, you can use `./cmd_linux.sh` to activate the env.

Thanks

**Comment by nbollman** [comment]
I assume you are using the one-click-installer... You may need to follow instructions for installing miniconda3 or making the conda a system environment variable, you'll need to activate the environment with 'conda acticvate path\to\env' folder that your text-generation-webui installs its environment to before running the 'pip install -r requirements.txt' inside the extension folder you want, or path to extension folder. Hope you can follow this mess of instructions and this helps!

**Reply by light-and-ray** [reply]
I use ./start_linux.sh

**Comment by nbollman** [comment]
When you run the start_windows.bat or start_linux.sh the program launches its /env with its own miniconda3, (I think it would make more sense to make miniconda3/anaconda a prerequisite for the program?)

**Comment by darklink4160** [comment]
This was never fixed at all. I think they should rename the one click installer if it's still this overly complicated for no reason. Couldn't this all be solved if they included this shit in the requirements installer? I think this is a form of gatekeeping and I find it disgusting.
